src/common/action.h

```diff
@@ -20,6 +20,10 @@
 
 #include <glib.h>
 
+#define DT_READ_ACTION_ONLY -FLT_MAX
+#define DT_ACTION_NOT_VALID -FLT_MAX
+#define DT_PERFORM_ACTION (move_size != DT_READ_ACTION_ONLY)
```

Comment:
Personally, I would have either had a parameterized macro instead of DT_PERFORM_ACTION or just written out the comparison where used.  The parameterless macro enforces the use of a specific variable name (that might be a good thing, or not) and hides the fact that we *are* checking the variable's value.

Reply:
Likewise.

Reply:
> that might be a good thing,

Yes, I consider that a good thing. The format of the `_action_process_` callbacks should all be as similar as possible without unnecessary individual interpretations that might mean that not all use cases are correctly supported. I tend to like to use macros to enforce such consistency. It is a taste thing.